src/thread.cpp

```diff
@@ -73,9 +73,10 @@ void Thread::clear() {
 /// Thread::start_searching() wakes up the thread that will start the search
 
 void Thread::start_searching() {
-
-  std::lock_guard<std::mutex> lk(mutex);
-  searching = true;
+  {
```

Comment:
can you explain the need / reason for this change?

Reply:
This can cost a little bit performance:

https://en.cppreference.com/w/cpp/thread/condition_variable/notify_one

> The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s); in fact doing so is a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock.

Also in their example they unlock before notifying:

```cpp
std::cout << "Notifying true change...\n";
lk.unlock();
cv.notify_one(); // waiting thread is notified with i == 1, cv.wait returns
```

I will add a comment

Reply:
No, it has nothing to do with code lines and I am only talking about unlocking, not locking.
Unlocking before notifying the other thread saves CPU-cycles/time. See the explanation I quoted/linked.

Reply:
```
void Thread::start_searching() {
  mutex.lock();
  searching = true;
  mutex.unlock();
  // Unlock before notifying saves a few CPU-cycles
  cv.notify_one();
}
```

Reply:
I see, I pretty misunderstood you. I changed as you suggested.